### ISSUE TITLE
Question] I want to project 3D point (X,Y,Z) onto image (x,y) corrdinate. Can you tell how can I do that?

### ISSUE THREAD

**Issue by ManishSahu53** [openMVG/openMVG#1192]
I have processed my images from openMVG and openMVS pipeline. Now I am trying to project a point (X,Y,Z) onto its pixel coordinate so I can get its rgb value. Can you tell me how to do it?

**Comment by ManishSahu53** [comment]
I found the answer x = K * (R * X + t)
x = pixel coordinate
K = intrinsic matrix
R = roatation
t = translation
X = 3D world point

**Comment by pmoulon** [comment]
You can use the projection function from the camera model:

`Vec2 image_projection = camera.project(pose(X));`

If the camera has distortion you can then compute the distorted image position
`Vec2 position = camera.get_d_pixel(image_projection);`

As explained in the doc for the pinhole camera model the projection is as you said: `x = K * (R * X + t)`
https://openmvg.readthedocs.io/en/latest/openMVG/cameras/cameras/

**Comment by ManishSahu53** [comment]
pmoulon,
I think there is something wrong here.
I have exported sfm.bin file to json and tried to manually calculate the pixel location from 3D point.
Location of 3D point = 707645.996417, 3442604.99292, 2285.00458226 
Camera Location =  707594.9015297502, 3442597.6518034075, 2367.264276279541

Its corresponding pixel coordinate in first camera is

1. From sfm.json file 
x= 2626.139892578125 y = 61.344600677490237 
2. I calculated from excel manually 
x= 2627.0968834059 y=  62.2335284256

Formula I used is 
x,y = undistort [K * R * (X-C)] # X and C are in UTM coordintae system

[sfm_data.zip](https://github.com/openMVG/openMVG/files/2140038/sfm_data.zip)
